src/common/w25x.h

```diff
@@ -21,8 +21,12 @@ extern "C" {
 /// This has to be called after the underlying SPI has been initialized
 /// and assigned using w25x_spi_assign.
 ///
-/// Returns true on success, false otherwise.
-extern bool w25x_init();
+/// @param init_DMA Dynamically allocate resources for DMA transfers
+///        DMA and RTOS facilities are not used if false,
+///        it has no effect if it was already allocated by previous call
+/// @retval true on success
+/// @retval false otherwise.
+extern bool w25x_init(bool init_DMA);
```

Comment:
/nitpick/

should be `init_dma`, as it uses snake_case...

Reply:
I am not a fan of this API change.

Lets first focus on whether it is actually needed. What is the motivation here.

Is it because `xEventGroupCreate` might not be safe to be called without the FreeRTOS running? We can detect such case using `xTaskGetSchedulerState() != taskSCHEDULER_RUNNING`.

Is it because `xEventGroupCreate` might not be safe to be called from within in ISR (like the hardfault)? We can detect that using `xPortIsInsideInterrupt()`.

Or is it because of something else? If just the two cases above are the reason for this parameter, I would get rid of the parameter and intiialize the DMA based on those checks.

Reply:
I want avoid dynamic RAM allocation in case w25x was not initialized before hardfault happened. w25x_init() is used in dump to xFlash and that is used in hard fault. DMA transfer is not used in hard fault, so I consider redundant to initialize it.

Idea behind this is that it might improve crash dump analysis if there would be no dynamic allocation done by dump itself.

Reply:
init_DMA renamed